adal/devApps/XFormsApp/SecondPage.cs

```diff
@@ -195,21 +251,62 @@ void LogCallback(LogLevel level, string message, bool containsPii)
                     this.result.Text += "Logs : " + DrainLogs();
                 });
             }
+        }
+
+        private async void AcquireTokenSilentWithBrokerButton_Clicked(object sender, EventArgs e)
+        {
+            this.result.Text = string.Empty;
+            AuthenticationContext ctx = new AuthenticationContext("https://login.microsoftonline.com/common");
+            string output = string.Empty;
+            try
+            {
+                AuthenticationResult result = await ctx.AcquireTokenSilentAsync("https://graph.microsoft.com", ClientIdBroker,
+                    new UserIdentifier(User, UserIdentifierType.OptionalDisplayableId), BrokerParameters).ConfigureAwait(false);
+                output = "Signed in User - " + result.UserInfo.DisplayableId;
+            }
+            catch (Exception exc)
+            {
+                output = exc.Message;
+            }
+            finally
+            {
+                Device.BeginInvokeOnMainThread(() =>
+                {
+                    this.result.Text += "Result : " + output;
+
+                    this.result.Text += "Logs : " + DrainLogs();
+                });
+            }
+        }
 
+        private string DeterminePlatformForRedirectUri()
+        {
+            switch (Device.RuntimePlatform)
+            {
+                case Device.iOS:
+                    RedirectURI = IOSBrokerRedirectURI;
+                    break;
+                case Device.Android:
+                    RedirectURI = AndroidBrokerRedirectURI;
+                    break;
+                default:
+                    throw new NotImplementedException();
+            }
+            return RedirectURI;
         }
 
-        private async void conditionalAccessButton_Clicked(object sender, EventArgs e)
+        private async void ConditionalAccessButton_Clicked(object sender, EventArgs e)
         {
             this.result.Text = string.Empty;
             AuthenticationContext ctx = new AuthenticationContext("https://login.microsoftonline.com/common");
             string output = string.Empty;
             string claims = "{\"access_token\":{\"polids\":{\"essential\":true,\"values\":[\"5ce770ea-8690-4747-aa73-c5b3cd509cd4\"]}}}";
-            
+
             try
             {
-                AuthenticationResult result = await ctx.AcquireTokenAsync("https://graph.windows.net", "<CLIENT_ID>",
-                        new Uri("adaliosapp://com.yourcompany.xformsapp"),
-                        Parameters, new UserIdentifier("<USER>", UserIdentifierType.OptionalDisplayableId), null, claims).ConfigureAwait(false);
+                AuthenticationResult result = await ctx.AcquireTokenAsync("https://graph.windows.net", ClientId,
+                        new Uri(IOSBrokerRedirectURI),
```

Comment:
is this intentional or did you mean to use the Device redirect uri?

Reply:
that's what was there before, so i just replaced it w/what it had before.